src/mlpack/tests/main_tests/linear_svm_test.cpp

```diff
@@ -0,0 +1,893 @@
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("labels", std::move(trainLabels));
+  SetInputParam("delta", double(1000));
+
+  // Second solution.
+  mlpackMain();
+
+  // Get the parameters of the output model obtained after second training.
+  const arma::mat &parameters2 =
+      CLI::GetParam<LinearSVM<>*>("output_model")->Parameters();
+
+  // Both solutions should be not equal.
+  CheckMatricesNotEqual(parameters1, parameters2);
+}
+
+/**
+  * Ensuring that no_intercept has some effects on the output.
+ **/
+BOOST_AUTO_TEST_CASE(LinearSVMDiffInterceptTest)
+{
+  arma::mat trainData;
+  if (!data::Load("iris.csv", trainData))
+    BOOST_FAIL("Cannot load test dataset iris.csv!");
+
+  arma::Row<size_t> trainLabels;
+  if (!data::Load("iris_labels.txt", trainLabels))
+    BOOST_FAIL("Cannot load test dataset iris_labels.txt!");
+
+  SetInputParam("training", trainData);
+  SetInputParam("labels", trainLabels);
+  SetInputParam("no_intercept", bool(false));
+
+  // First solution.
+  mlpackMain();
+
+  // Get the parameters of the output model obtained after first training.
+  const arma::mat parameters1 =
+      std::move(CLI::GetParam<LinearSVM<>*>("output_model")
+                ->Parameters());
+
+  // Reset the settings.
+  bindings::tests::CleanMemory();
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("labels", std::move(trainLabels));
+  SetInputParam("no_intercept", bool(true));
+
+  // Second solution.
+  mlpackMain();
+
+  // Get the parameters of the output model obtained after second training.
+  const arma::mat &parameters2 =
+      CLI::GetParam<LinearSVM<>*>("output_model")->Parameters();
+
+  // Both solutions should be not equal.
+  CheckMatricesNotEqual(parameters1, parameters2);
+}
+
+/**
+  * Ensuring that number of classes has some effects on the output.
+ **/
+BOOST_AUTO_TEST_CASE(LinearSVMDiffNumberOfClassesTest)
+{
+  arma::mat trainData;
+  if (!data::Load("iris.csv", trainData))
+    BOOST_FAIL("Cannot load test dataset iris.csv!");
+
+  arma::Row<size_t> trainLabels;
+  if (!data::Load("iris_labels.txt", trainLabels))
+    BOOST_FAIL("Cannot load test dataset iris_labels.txt!");
+
+  SetInputParam("training", trainData);
+  SetInputParam("labels", trainLabels);
+  SetInputParam("number_of_classes", int(0));
+
+  // First solution.
+  mlpackMain();
+
+  // Get the parameters of the output model obtained after first training.
+  const arma::mat parameters1 =
+      std::move(CLI::GetParam<LinearSVM<>*>("output_model")
+                ->Parameters());
+
+  // Reset the settings.
+  bindings::tests::CleanMemory();
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("labels", std::move(trainLabels));
+  SetInputParam("number_of_classes", int(2));
```

Comment:
I think this test is unsafe---if we specify `number_of_classes` less than the actual number of classes, then there are likely to be invalid memory accesses.  On the other hand, if it is specified to be *more* than the actual number of classes, it should work fine without invalid memory accesses, but the parameters associated with the extra classes may take any value.

Reply:
Removed.